g600.c

```diff
@@ -0,0 +1,149 @@
+  [16] = "xdotool key Return", // G17
+  [17] = "i3-msg fullscreen", // G18
+  [18] = "xdotool key ctrl+slash t", // G19
+  [19] = "", // G20
+  [20] = "xdotool key alt+Left", // G-shift + scroll left
+  [21] = "xdotool key alt+Right", // G-shift + scroll right
+  [22] = "xdotool key ctrl+v", // G-shift + G8
+  [23] = "xdotool key ctrl+shift+v", // G-shift + G7
+  [24] = "i3-msg workspace prev_on_output", // G-shift + G9
+  [25] = "i3-msg move workspace prev_on_output", // G-shift + G10
+  [26] = "i3-msg kill", // G-shift + G11
+  [27] = "pulseaudio-ctl up", // G-shift + G12
+  [28] = "pulseaudio-ctl mute", // G-shift + G13
+  [29] = "xdotool key ctrl+shift+z ctrl+y", // G-shift + G14
+  [30] = "xdotool key Home", // G-shift + G15
+  [31] = "xdotool key ctrl+Home", // G-shift + G16
+  [32] = "xdotool key Escape", // G-shift + G17
+  [33] = "i3-msg fullscreen", // G-shift + G18
+  [34] = "", // G-shift + G19
+  [35] = "", // G-shift + G20
+  [37] = "echo button down"
+};
+const char *upCommands[] = {
+  //[scancode] = "command to run",
+  [37] = "echo button up"
+};
+
+int starts_with(const char* haystack, const char* prefix) {
+  size_t prefix_length = strlen(prefix), haystack_length = strlen(haystack);
+  if (haystack_length < prefix_length) return 0;
+  return strncmp(prefix, haystack, prefix_length) == 0;
+}
+
+int ends_with(const char* haystack, const char* suffix) {
+  size_t suffix_length = strlen(suffix), haystack_length = strlen(haystack);
+  if (haystack_length < suffix_length) return 0;
+  size_t haystack_end = haystack + haystack_length - suffix_length;
+  return strncmp(suffix, haystack_end, suffix_length) == 0;
+}
+
+// Returns non-0 on error.
+int find_g600(char *path) {
+  //*path = kDir;
+  DIR *dir;
+  struct dirent *ent;
+  if (!(dir = opendir(kDir))) {
+    return 1;
+  }
+  while ((ent = readdir(dir))) {
+    if (starts_with(ent->d_name, kPrefix) && ends_with(ent->d_name, kSuffix)) {
+      strcpy(path, kDir);
+      strcat(path, ent->d_name);
+
+      printf("full path is %s\n", path);
+
+      //*path += ent->d_name;
+      closedir(dir);
+      return 0;
+    }
+  }
+  closedir(dir);
+  return 2;
+}
+
+int main() {
+  printf("Starting G600 Linux controller.\n\n");
+  printf("It's a good idea to configure G600 with Logitech Gaming Software before running this program:\n");
+  printf(" - assign left, right, middle mouse button and vertical mouse wheel to their normal functions\n");
+  printf(" - assign the G-Shift button to \"G-Shift\"\n");
+  printf(" - assign all other keys (including horizontal mouse wheel) to arbitrary (unique) keyboard keys\n");
+  printf("\n");    
+  char path[1024];
+  int find_error = find_g600(&path);
+  if (find_error) {
+    printf("Error: Couldn't find G600 input device.\n");
+    switch(find_error) {
+    case 1: 
+      printf("Suggestion: Maybe the expected directory (%s) is wrong. Check whether this directory exists and fix it by editing \"g600.cpp\".\n", kDir);
```

Comment:
Thanks for the PR. Making this C is sounds right :)

Could you update the filename in the error message?